src/libtiled/hexagonalrenderer.cpp

```diff
@@ -338,14 +342,14 @@ void HexagonalRenderer::drawTileLayer(QPainter *painter,
         if (p.doStaggerY(startTile.y() + layer->y()))
             startPos.rx() -= p.columnWidth;
 
-        for (; startPos.y() < rect.bottom() && startTile.y() < layer->height(); startTile.ry()++) {
+        for (; startPos.y() < rect.bottom() && (startTile.y() < layer->height() || map()->infinite()); startTile.ry()++) {
```

Comment:
Please still optimize this to use an `endY` that is either `layer->height()` or `layer->bounds()`, same for X.